permissions/src/androidLibMain/kotlin/AndroidPermissionsManager.kt

```diff
@@ -143,18 +184,26 @@ class AndroidPermissionsManager<P : Permission> constructor(
      * `true` if the permission has been granted.
      */
     internal val hasPermissions: Boolean get() {
-        return permissions.fold(true) { previous, permission ->
-            when (ContextCompat.checkSelfPermission(context, permission)) {
-                PackageManager.PERMISSION_DENIED -> false
-                PackageManager.PERMISSION_GRANTED -> true
-                else -> false
+        return filteredPermissionsStates.values.fold(true) { previous, permissionState ->
+            when (permissionState) {
+                AndroidPermissionState.GRANTED -> true
+                AndroidPermissionState.DENIED -> false
+                AndroidPermissionState.DENIED_DO_NOT_ASK -> false
             } && previous
         }
     }
 
-    private fun updateLastPermissions() {
+    private fun updatePermissionsStates() {
         permissions.forEach {
-            lastPermission[it] = ContextCompat.checkSelfPermission(context, it)
+            val oldPermissionState = permissionsStates[it]
+            val newPermissionState = AndroidPermissionState.get(context, it)
+            if (oldPermissionState == AndroidPermissionState.DENIED_DO_NOT_ASK
+                && newPermissionState == AndroidPermissionState.DENIED) {
+                return
```

Comment:
It would be nice to have a comment here in the code to explain why this is done.

Reply:
Good point, I'll add an explanation.